lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1122,18 +1122,18 @@ export class UserAgentApplication {
     const account: Account = accountObj || this.getAccount();
 
     // This is a final check for all queryParams added so far; preference order: sid > login_hint
-    // sid cannot be passed along with login_hint, hence we check both are not populated yet in queryParameters so far
-    if (account) {
+    // sid cannot be passed along with login_hint or domain_hint, hence we check both are not populated yet in queryParameters so far
```

Comment:
The comment says sid cannot be passed along with login_hint or domain_hint, but where do we actually check this and remove the hint if sid is available?